Bug: Cluster-api-provider-vsphere crashes in case the cluster's providerStatus is not yet set

Observed a controller crash with the following trace:

Reason seems to be that the method `vsphereutils.GetClusterProviderStatus(cluster)` can possibly return `nil, nil` which is not handled properly in the code https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/f67fa04674f2aefc620f1157e51723b8ad30e255/pkg/cloud/vsphere/provisioner/govmomi/create.go#L561

Causing the below crash when `clusterstatus` is returned nil. Nil case needs to be treated in the same way as the cluster is `NotReady`
This can happen, if the machine reconcile loop is ahead of the cluster reconcile loop (as evident in the trace below which can happen) 

```
I0319 17:58:16.666083       1 main.go:70] Starting the Cmd.
I0319 17:58:16.667078       1 reflector.go:131] Starting reflector *v1alpha1.Machine (2m0s) from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126
I0319 17:58:16.667310       1 reflector.go:169] Listing and watching *v1alpha1.Machine from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126
I0319 17:58:16.667352       1 reflector.go:131] Starting reflector *v1alpha1.Cluster (2m0s) from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126
I0319 17:58:16.667446       1 reflector.go:169] Listing and watching *v1alpha1.Cluster from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126
I0319 17:58:16.668159       1 round_trippers.go:438] GET https://100.16.0.1:443/apis/cluster.k8s.io/v1alpha1/clusters?resourceVersion=3233&timeout=8m1s&timeoutSeconds=481&watch=true 200 OK in 1 milliseconds
I0319 17:58:16.669167       1 round_trippers.go:438] GET https://100.16.0.1:443/apis/cluster.k8s.io/v1alpha1/machines?limit=500&resourceVersion=0 200 OK in 1 milliseconds
I0319 17:58:16.737508       1 round_trippers.go:438] GET https://100.16.0.1:443/apis/cluster.k8s.io/v1alpha1/clusters?limit=500&resourceVersion=0 200 OK in 1 milliseconds
I0319 17:58:16.737521       1 round_trippers.go:438] GET https://100.16.0.1:443/apis/cluster.k8s.io/v1alpha1/machines?resourceVersion=3248&timeoutSeconds=582&watch=true 200 OK in 0 milliseconds
I0319 17:58:16.737897       1 round_trippers.go:438] GET https://100.16.0.1:443/apis/cluster.k8s.io/v1alpha1/machines?resourceVersion=3248&timeout=8m19s&timeoutSeconds=499&watch=true 200 OK in 1 milliseconds
I0319 17:58:16.738912       1 round_trippers.go:438] GET https://100.16.0.1:443/apis/cluster.k8s.io/v1alpha1/clusters?resourceVersion=3233&timeoutSeconds=431&watch=true 200 OK in 1 milliseconds
I0319 17:58:16.766669       1 shared_informer.go:123] caches populated
I0319 17:58:16.866843       1 shared_informer.go:123] caches populated
I0319 17:58:16.866991       1 controller.go:113] Reconciling Machine "worker-vvl697rrjg"
I0319 17:58:16.867030       1 shared_informer.go:123] caches populated
I0319 17:58:16.867232       1 controller.go:89] Running reconcile Cluster for cluster-bh96bz5ld5
...
...
E0319 17:58:16.964839       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:388
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:561
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:447
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:138
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:67
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/machineactuator.go:64
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:214
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x13b27a7]

goroutine 218 [running]:
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x15e8c60, 0x27b5760)
	/usr/local/go/src/runtime/panic.go:502 +0x229
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi.(*Provisioner).getStartupScript(0xc4207c0f50, 0xc4208e29c0, 0xc420277340, 0x2, 0x0, 0x0, 0x1742360)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:561 +0x4d7
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi.(*Provisioner).getCloudInitUserData(0xc4207c0f50, 0xc4208e29c0, 0xc420277340, 0x1974390, 0x19742a0, 0xc4202a7e30, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:447 +0x5d
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi.(*Provisioner).cloneVirtualMachine(0xc4207c0f50, 0xc420512520, 0xc4208e29c0, 0xc420277340, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:138 +0xab
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi.(*Provisioner).Create(0xc4207c0f50, 0x1a61420, 0xc4200b2028, 0xc4208e29c0, 0xc420277340, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/create.go:67 +0x315
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere.(*VsphereClient).Create(0xc4207dc900, 0x1a61420, 0xc4200b2028, 0xc4208e29c0, 0xc420277340, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/machineactuator.go:64 +0x6f
sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc420693840, 0xc4206042f0, 0x9, 0xc4205f46c0, 0x11, 0x27c8cc0, 0xc42011baec, 0xc420058560, 0x554356)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:214 +0x8d6
sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc42017e960, 0x0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x188
sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4202020a0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4202020a0, 0x3b9aca00, 0x0, 0x1, 0xc4200a51a0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4202020a0, 0x3b9aca00, 0xc4200a51a0)
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x35b
```

--- a/pkg/cloud/vsphere/provisioner/govmomi/create.go
+++ b/pkg/cloud/vsphere/provisioner/govmomi/create.go
@@ -558,7 +558,7 @@ func (pv *Provisioner) getStartupScript(cluster *clusterv1.Cluster, machine *clu
 			klog.Infof("Error fetching cluster ProviderStatus field: %s", err)
 			return "", err
 		}
-		if clusterstatus.APIStatus != vsphereconfigv1.ApiReady {
+		if clusterstatus == nil || clusterstatus.APIStatus != vsphereconfigv1.ApiReady {
 			duration := vsphereutils.GetNextBackOff()
 			klog.Infof("Waiting for Kubernetes API Status to be \"Ready\". Retrying in %s", duration)
 			return "", &clustererror.RequeueAfterError{RequeueAfter: duration}

